src/core/reactor.cc

```diff
@@ -4228,6 +4316,19 @@ class disk_config_params {
             cfg.req_count_rate = io_queue::read_request_base_count * (unsigned long)per_io_group(p.read_req_rate, nr_groups);
             cfg.disk_req_write_to_read_multiplier = (io_queue::read_request_base_count * p.read_req_rate) / p.write_req_rate;
         }
+
+        // Use the full cost model if io-properties are in use and min and max IOP size are available
+        if (p.read_req_rate != std::numeric_limits<uint64_t>::max() &&
+            p.read_bytes_rate != std::numeric_limits<uint64_t>::max() &&
+            p.min_iop_size && p.max_iop_size) {
+
+            std::tie(cfg.write_iop_factor, cfg.write_throughput_factor) = calculate_full_cost_formula_factors(true, p);
+            std::tie(cfg.read_iop_factor, cfg.read_throughput_factor) = calculate_full_cost_formula_factors(false, p);
```

Comment:
Please, note, there's "rate_factor" thing in config, which now interferes with these new factors and make things very complicated. Like this

First, we have iops_max and bandwidth_max -- the points at which the "diskplorer line" crosses the x-y axis.
Then the rate_factor is applied this shifting (scaling?) the line along the 45-degrees vector.
Now new parameters are applied and shift the individual corner points each by its own factor.

So far I've no good ideas what to do about it, it's just an observation

Reply:
After a bit more thinking, I recommend making rate factor and new parameters mutually exclusive. I.e. -- when both are present in the io-properties, the mountpoint_params creator code should throw configuration exception. They both tune that "famous" line on the iops:bw plane, and should be two different ways of doing it, not two incremental updates to the default line.